compiler/src/org.graalvm.compiler.virtual/src/org/graalvm/compiler/virtual/phases/ea/VirtualizerToolImpl.java

```diff
@@ -309,6 +310,10 @@ public void createVirtualObject(VirtualObjectNode virtualObject, ValueNode[] ent
         closure.addVirtualAlias(virtualObject, virtualObject);
         PartialEscapeClosure.COUNTER_ALLOCATION_REMOVED.increment(debug);
         effects.addVirtualizationDelta(1);
+        if (sourcePosition != null) {
+            assert virtualObject.getNodeSourcePosition() == null : "source pos already set!";
```

Comment:
We might add `|| sourcePosition == virtualObject.getNodeSourcePosition()` if we decide to pass `virtualObject.getNodeSourcePosition()` instead of null.

Reply:
Yes, will do.